### ISSUE TITLE
Feature Request: New SSL algorithm called Maximum Manifold Capacity Representations

### ISSUE THREAD

**Issue by RylanSchaeffer** [lightly-ai/lightly#1423]
This is a NeurIPS 2023 paper with a new SSL vision method: https://neurips.cc/virtual/2023/poster/70447

It'd be great if Lightly could add it.

**Comment by guarin** [comment]
This looks interesting, will have a closer look later :)

**Comment by RylanSchaeffer** [comment]
Sounds good!

The loss is quite straightforward. Suppose `x` has shape `(batch size, number of views/transformations e.g., 2, 4, 32, projection output dimension`. MMCR proposes the following loss:

```
avg_over_views = torch.mean(x, dim=1)
loss = - torch.linalg.matrix_norm(avg_over_views, ord="nuc")
```

**Comment by guarin** [comment]
Arxiv link: https://arxiv.org/abs/2303.03307

**Comment by guarin** [comment]
Remaining issues that we have to do before this issue is complete:
* #1451 
* #1452
